File: elixir/ch1/ex115.ex
# cd("c:/Users/Yohan/IdeaProjects/sicp/elixir/ch1")
# Load with import_file("filename")
# Run with Module.func(args)
defmodule Ex115 do
  @moduledoc false
  def cube(x), do: x * x * x

  def p(x), do: 3 * x - 4 * cube x

  def sine(angle) when (abs angle) < 0.1, do: angle
  def sine(angle), do: p (sine (angle / 3.0))

end

File: julia/ch1/Ex115.jl
"""
# module Ex114


# Examples

```jldoctest
julia> include("c:/Users/Yohan/IdeaProjects/sicp/julia/ch1/exXXX.jl")

julia> ExXXX.func(arg)
```
"""
module Ex115

cube(x) = x * x * x

p(x) = 3 * x - 4 * cube(x)

function sine(angle)
    if abs(angle) < 0.1
        return angle
    else
        return p(sine(angle / 3.0))
    end
end

end
